Sources/GRPCHTTP2Core/Internal/NIOChannelPipeline+GRPC.swift

```diff
@@ -51,10 +51,23 @@ extension ChannelPipeline.SynchronousOperations {
     )
     try self.addHandler(flushNotificationHandler)
 
+    // Window size which mustn't exceed 2^31 - 1 (RFC 9113 § 6.5.2).
+    let clampedTargetWindowSize = min(http2Config.targetWindowSize, (1 << 31) - 1)
+
+    // Max frame size must be in the range 2^14 ..< 2^24 (RFC 9113 § 4.2).
+    let clampedMaxFrameSize: Int
+    if http2Config.maxFrameSize >= (1 << 24) {
+      clampedMaxFrameSize = (1 << 24) - 1
+    } else if http2Config.maxFrameSize < (1 << 14) {
+      clampedMaxFrameSize = (1 << 14)
+    } else {
+      clampedMaxFrameSize = http2Config.maxFrameSize
+    }
```

Comment:
Presumably this is all repeated in the client? Might be worth extracting into a couple of internal `validateHTTP2<Thing>` functions